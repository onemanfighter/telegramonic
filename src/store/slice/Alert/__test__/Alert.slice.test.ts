import { AlertComponentState, appStore } from '@store';
import { act, renderHook } from '@testing-library/react';

jest.useFakeTimers();

describe('Alert slice', () => {
  const alertData: AlertComponentState = {
    description: 'description',
    showIcon: false,
    title: 'title',
    status: 'success',
    variant: 'solid',
  };
  it('should return alert state', () => {
    const alertState = renderHook(() => appStore()).result.current;

    expect(alertState).toMatchSnapshot();
  });

  it('should return alert state with initial value', () => {
    const alertState = renderHook(() => appStore()).result.current;

    expect(alertState.Alert.alert).toEqual({
      description: '',
      showIcon: true,
      title: '',
      status: undefined,
      variant: 'subtle',
    });
  });

  it('should return alert state on alert data set using setAlertStateData', () => {
    const { result } = renderHook(() => appStore());

    act(() => {
      result.current.Alert.setAlertStateData(alertData);
    });

    expect(result.current.Alert.alert).toMatchSnapshot();
  });

  it('setAlertVisible should update the visible state of the alert state', () => {
    const { result } = renderHook(() => appStore());

    act(() => {
      result.current.Alert.setAlertVisible(true);
    });

    expect(result.current.Alert.alertVisibleState).toBeTruthy();

    act(() => {
      result.current.Alert.setAlertVisible(false);
    });

    expect(result.current.Alert.alertVisibleState).toBeFalsy();
  });
});
