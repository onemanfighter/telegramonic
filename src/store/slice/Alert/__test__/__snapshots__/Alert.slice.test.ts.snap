// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert slice should return alert state 1`] = `
{
  "Alert": {
    "alert": {
      "description": "",
      "showIcon": true,
      "status": undefined,
      "title": "",
      "variant": "subtle",
    },
    "alertVisibleState": false,
    "setAlertStateData": [Function],
    "setAlertVisible": [Function],
  },
  "Drawer": {
    "closeDrawer": [Function],
    "drawerData": {
      "children": null,
      "footer": null,
      "isOpen": false,
      "onModalClose": [Function],
      "title": "",
    },
    "openDrawer": [Function],
    "openState": "CLOSE",
  },
  "Modal": {
    "closeModal": [Function],
    "modalData": {
      "message": "",
      "onModalClose": [Function],
      "onPrimaryButtonClick": [Function],
      "primaryButtonLabel": "",
      "title": "",
    },
    "openModal": [Function],
    "openState": "CLOSE",
  },
}
`;

exports[`Alert slice should return alert state on alert data set using setAlertStateData 1`] = `
{
  "description": "description",
  "showIcon": false,
  "status": "success",
  "title": "title",
  "variant": "solid",
}
`;
